Pocket/pocket-ios
# IN-661: Save From Clipboard

## Summary
Support Save From Clipboard by Adding a Banner that allows user to save an item after copying a URL

## References 
IN-661, IN-662, IN-664

## Implementation Details
Add `BannerView` to host the main UI element for the banner and the data to populate the UI elements are handled by `SaveFromClipboardViewModel`. Since we want this banner to appear in various places in the app, the logic to handle animation and presentation is located in `RootCoordinator`.

## Test Steps
- [x] Given a user has a URL copied to their clipboard,
and the url is not already saved,
and the url is not the last URL presented,
when the app is launched,
then I should see a banner asking to save the URL.

- [x] Given a user has a URL copied to their clipboard,
and the url is not already saved,
and the url is the last URL presented,
when the app is launched,
then I should see the banner asking to save the URL.

- [x] Given a user has a URL copied to their clipboard,
and the url is already saved,
when the app is launched,
then I should see the banner asking to save the URL.

- [x] Given there is a url on the clipboard,
and the clipboard banner has been presented,
when the “Save” button is tapped,
the URL should be saved.

- [x] Given there is a url on the clipboard,
and the clipboard banner has been presented,
when the “Save” button is tapped,
the URL should appear in the user’s list.

- [x] Given there is a url on the clipboard,
and the clipboard banner has been presented,
when the “Save” button is tapped,
and the URL save has been synced with the server,
the list should replace the url with the saved item.

- [x] Given the clipboard banner is present,
when the user swipes down on the banner,
the banner should dismiss.

Note: For now, it is okay for the user to continue to see the banner. In the case of the banner, if the user already saved the item - then it will bring the item to the top. 

## PR Checklist:

## Screenshots

https://user-images.githubusercontent.com/6743397/183942732-eda575bc-4fac-471c-b961-a369e02bcf71.mp4
